### PR TITLE
removed a dead link from fine-tuning.md in docs

### LINKED ISSUE #17619: [Documentation]: Dead Link in https://docs.llamaindex.ai/en/stable/optimizing/fine-tuning/fine-tuning/#fine-tuning-llama-2-for-better-text-to-sql
### Documentation Issue Description

Image

In the above section of the documentation in fine-tuning, the first link returns a 404 error. I can fix this issue, If you guys want to assign this to me.

Image

### Documentation Link

https://docs.llamaindex.ai/en/stable/optimizing/fine-tuning/fine-tuning/#fine-tuning-llama-2-for-better-text-to-sql

### PR DESCRIPTION
**Pull request by Riddhimaan-Senapati**
# Description

Removed a dead link in the fine-tuning.md in docs as the file is no longer present in the repository. This is the link that was removed: https://docs.llamaindex.ai/en/stable/optimizing/examples/finetuning/gradient/gradient_fine_tuning.ipynb

Fixes #17619

## New Package?

Did I fill in the `tool.llamahub` section in the `pyproject.toml` and provide a detailed README.md for my new integration or package?

- [ ] Yes
- [x] No

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [ ] Yes
- [x] No

## Type of Change

Please delete options that are not relevant.

- [x] This change requires a documentation update

## How Has This Been Tested?

Your pull-request will likely not be merged unless it is covered by some form of impactful unit testing.

- [ ] I added new unit tests to cover this change
- [x] I believe this change is already covered by existing unit tests

## Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added Google Colab support for the newly added notebooks.
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

**Comment by Riddhimaan-Senapati** [comment]
@tobi @wpiekutowski @xevo @bborn @logan-markewich Can anyone of you review my PR?

**Review by logan-markewich — APPROVED** [review]
